ouroboros-consensus-cardano/tools/db-analyser/Analysis.hs

```diff
@@ -245,38 +249,50 @@ decreaseLimit (Limit 0) = Nothing
 decreaseLimit (Limit n) = Just . Limit $ n - 1
 
 processAll ::
-     forall blk b st. HasHeader blk
+     forall blk b st. (HasHeader blk, HasAnnTip blk)
   => Either (ImmutableDB IO blk) (ChainDB IO blk)
   -> ResourceRegistry IO
   -> BlockComponent blk b
+  -> ExtLedgerState blk
   -> Limit
   -> st
   -> (st -> b -> IO st)
   -> IO st
 processAll = either processAllImmutableDB processAllChainDB
 
 processAll_ ::
-     forall blk b. HasHeader blk
+     forall blk b. (HasHeader blk, HasAnnTip blk)
   => Either (ImmutableDB IO blk) (ChainDB IO blk)
   -> ResourceRegistry IO
   -> BlockComponent blk b
+  -> ExtLedgerState blk
   -> Limit
   -> (b -> IO ())
   -> IO ()
-processAll_ db rr blockComponent limit callback =
-    processAll db rr blockComponent limit () (const callback)
+processAll_ db registry blockComponent initLedger limit callback =
+    processAll db registry blockComponent initLedger limit () (const callback)
 
 processAllChainDB ::
-     forall st blk b. HasHeader blk
+     forall st blk b. (HasHeader blk, HasAnnTip blk)
   => ChainDB IO blk
   -> ResourceRegistry IO
   -> BlockComponent blk b
+  -> ExtLedgerState blk
   -> Limit
   -> st
   -> (st -> b -> IO st)
   -> IO st
-processAllChainDB chainDB rr blockComponent limit initState callback = do
-    itr <- ChainDB.streamAll chainDB rr blockComponent
+processAllChainDB chainDB registry blockComponent (ExtLedgerState {..}) limit initState callback = do
```

Comment:
This function no longer needs this, right? We decided (at least for now) in that other comment thread, that the ChainDB option does not support an `--analyse-from`.

Reply:
EG Move the new `ExtLedgerState blk` to `processAll` and `processAll_` into the `Left` side of the `Either`, paired with the `ImmutableDB`.

Reply:
Better yet: leave `processAll` alone, and introduce a new `process***` combinator for analyses that need an initial ledger state. That combinator would eg immediately always fail if `db` is a `Right`.

Reply:
If this refactor is too intensive, feel free to open a tech-debt follow-up issue to unblock this PR. My request here is an implementation detail. (Unlike the other thread where I requested an explicit check on the command-line arguments' coherency.)

Reply:
I'm ok implementing it now. I'm only hesitant because I have a gut feeling that the ledger and/or benchmarking team will explicitly ask for `--analyse-from` for ChainDB. But we can always revisit it I guess (even reverting the single commit if needed)

Reply:
After thinking about it, how about we create a tech-debt ticket but we will implement it in a week or two from now, once the tool is used by other teams and they will confirm that they like what they have?

Reply:
I've created two tickets:
1. https://input-output.atlassian.net/browse/CAD-3477
2. https://input-output.atlassian.net/browse/CAD-3516